src/realm-settings/RSAModal.tsx

```diff
@@ -43,31 +43,44 @@ export const RSAModal = ({
   const { addAlert } = useAlerts();
   const { handleSubmit, control } = useForm({});
   const [isRSAalgDropdownOpen, setIsRSAalgDropdownOpen] = useState(false);
-  const [displayName, setDisplayName] = useState("");
-  const realm = useRealm();
+
+  const { id } = useParams<{ id: string }>();
 
   const [keyFileName, setKeyFileName] = useState("");
   const [certificateFileName, setCertificateFileName] = useState("");
 
   const allComponentTypes =
-    serverInfo.componentTypes!["org.keycloak.keys.KeyProvider"];
+    serverInfo.componentTypes?.["org.keycloak.keys.KeyProvider"] ?? [];
 
   const save = async (component: ComponentRepresentation) => {
     try {
-      await adminClient.components.create({
-        parentId: realm.realm,
-        name: displayName !== "" ? displayName : providerType,
-        providerId: providerType,
-        providerType: "org.keycloak.keys.KeyProvider",
-        ...component,
-      });
-      refresh!();
-      addAlert(t("realm-settings:saveProviderSuccess"), AlertVariant.success);
-      handleModalToggle!();
+      if (id) {
+        await adminClient.components.update(
+          { id },
+          {
+            ...component,
+            parentId: component.parentId,
+            providerId: providerType,
+            providerType: "org.keycloak.keys.KeyProvider",
+          }
+        );
+        addAlert(t("realm-settings:saveProviderSuccess"), AlertVariant.success);
+      } else {
+        await adminClient.components.create({
+          ...component,
+          parentId: component.parentId,
+          providerId: providerType,
+          providerType: "org.keycloak.keys.KeyProvider",
+        });
+        handleModalToggle!();
```

Comment:
Using the non-null assertion operator can lead to runtime errors, in this case `handleModalToggle` is always provided upstream so it can be made required.

```suggestion
        handleModalToggle();
```